src/Engine/Segments/SegmentEvaluator.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace Flagsmith\Engine\Segments;
+
+use Flagsmith\Engine\Environments\EnvironmentModel;
+use Flagsmith\Engine\Identities\IdentityModel;
+use Flagsmith\Engine\Identities\Traits\TraitModel;
+use Flagsmith\Engine\Utils\HashingTrait;
+
+class SegmentEvaluator
+{
+    use HashingTrait;
+    /**
+     * Get the identity segments.
+     * @param EnvironmentModel $environment
+     * @param IdentityModel $identity
+     * @param array|null $overrideTraits
+     * @return array
+     */
+    public static function getIdentitySegments(
+        EnvironmentModel $environment,
+        IdentityModel $identity,
+        array $overrideTraits = null
+    ): array {
+        return array_filter(
+            $environment->getProject()->getSegments()->getArrayCopy(),
+            fn (SegmentModel $segment) => self::evaluateIdentityInSegment(
+                $identity,
+                $segment,
+                $overrideTraits
+            )
+        );
+    }
+
+    /**
+     * evaluate identity in segment.
+     * @param IdentityModel $identity
+     * @param SegmentModel $segment
+     * @param array|null $overrideTraits
+     * @return bool
+     */
+    public static function evaluateIdentityInSegment(
+        IdentityModel $identity,
+        SegmentModel $segment,
+        array $overrideTraits = null
+    ): bool {
+        $rulesCount = count($segment->getRules());
+
+        if (empty($rulesCount)) {
+            return false;
+        }
+
+        $traitSegmentRules = [];
+        foreach ($segment->getRules() as $rule) {
+            $segmentRule = self::traitsMatchSegmentRule(
+                $overrideTraits ?? $identity->getIdentityTraits()->getArrayCopy(),
+                $rule,
+                $segment->getId(),
+                $identity->compositeKey()
+            );
+
+            $traitSegmentRules[] = $segmentRule;
+
+            if (!$segmentRule) {
+                break;
+            }
+        }
+
+        return self::all($traitSegmentRules);
+    }
+
+    /**
+     * traits match segment rule.
+     * @param array $identityTraits
+     * @param SegmentRuleModel $rule
+     * @param mixed $segmentId
+     * @param mixed $identityId
+     * @return bool
+     */
+    private static function traitsMatchSegmentRule(
+        array $identityTraits,
+        SegmentRuleModel $rule,
+        $segmentId,
+        $identityId
+    ): bool {
+        $traitConditions = [true];
+        $conditions = $rule->getConditions()->getArrayCopy();
+
+        if (count($conditions) > 0) {
+            $traitConditions = array_map(fn ($condition) => self::traitsMatchSegmentCondition(
+                $identityTraits,
+                $condition,
+                $segmentId,
+                $identityId
+            ), $conditions);
+        }
+
+        $matchesCondition = $rule->matchingFunction()($traitConditions);
+
+        $segmentRules = array_map(
```

Comment:
Again, I think this variable could be better named, right? I don't think it's an array of `SegmentRuleModel`s?

Reply:
This was matching segment rules on traits. Renamed to $traitsMatchingSegmentRules.